test/hotspot/jtreg/compiler/vectorization/TestRoundVectorFloatAll.java

```diff
@@ -0,0 +1,128 @@
+ * @summary Test vector intrinsic for Math.round(float) in full 32 bits range
+ *
+ * @library /test/lib /
+ * @modules java.base/jdk.internal.math
+ * @run main compiler.vectorization.TestRoundVectorFloatRandom -XX:-TieredCompilation -XX:CompileThresholdScaling=0.3
+ * @run main compiler.vectorization.TestRoundVectorFloatRandom -XX:-TieredCompilation -XX:CompileThresholdScaling=0.3 -XX:MaxVectorSize=8
+ * @run main compiler.vectorization.TestRoundVectorFloatRandom -XX:-TieredCompilation -XX:CompileThresholdScaling=0.3 -XX:MaxVectorSize=16
+ * @run main compiler.vectorization.TestRoundVectorFloatRandom -XX:-TieredCompilation -XX:CompileThresholdScaling=0.3 -XX:MaxVectorSize=32
+ */
+
+package compiler.vectorization;
+
+import jdk.internal.math.FloatConsts;
+
+public class TestRoundVectorFloatAll {
+  private static final int ITERS  = 11000;
+  private static final int ARRLEN = 997;
+  private static final float ADD_INIT = -7500.f;
+
+  public static void main(String args[]) {
+    test();
+  }
+
+  static int golden_round(float a) {
+    // below code is copied from java.base/share/classes/java/lang/Math.java
+    //  public static int round(float a) { ... }
+
+    int intBits = Float.floatToRawIntBits(a);
+    int biasedExp = (intBits & FloatConsts.EXP_BIT_MASK)
+            >> (FloatConsts.SIGNIFICAND_WIDTH - 1);
+    int shift = (FloatConsts.SIGNIFICAND_WIDTH - 2
+            + FloatConsts.EXP_BIAS) - biasedExp;
+    if ((shift & -32) == 0) { // shift >= 0 && shift < 32
+        // a is a finite number such that pow(2,-32) <= ulp(a) < 1
+        int r = ((intBits & FloatConsts.SIGNIF_BIT_MASK)
+                | (FloatConsts.SIGNIF_BIT_MASK + 1));
+        if (intBits < 0) {
+            r = -r;
+        }
+        // In the comments below each Java expression evaluates to the value
+        // the corresponding mathematical expression:
+        // (r) evaluates to a / ulp(a)
+        // (r >> shift) evaluates to floor(a * 2)
+        // ((r >> shift) + 1) evaluates to floor((a + 1/2) * 2)
+        // (((r >> shift) + 1) >> 1) evaluates to floor(a + 1/2)
+        return ((r >> shift) + 1) >> 1;
+    } else {
+        // a is either
+        // - a finite number with abs(a) < exp(2,FloatConsts.SIGNIFICAND_WIDTH-32) < 1/2
+        // - a finite number with ulp(a) >= 1 and hence a is a mathematical integer
+        // - an infinity or NaN
+        return (int) a;
+    }
+  }
+
+  static int test() {
+    float[] input = new float[ARRLEN];
+    int [] res = new int[ARRLEN];
+
+    for (int i=0; i<ARRLEN; i++) {
+      float val = ADD_INIT+(float)i;
+      input[i] = val;
+    }
+    // Warmup
+    System.out.println("Warmup");
+    for (int i=0; i<ITERS; i++) {
+      test_round(res, input);
+    }
+
+    // Test and verify results
+    System.out.println("Verification");
+    int errn = 0;
+    for (long l = Integer.MIN_VALUE; l <= Integer.MAX_VALUE; l+=ARRLEN) {
+      for (int i = 0; i < ARRLEN; i++) {
+        input[i] = (int)(l+i);
+      }
```

Comment:
What is this array for? As far as i can tell it does nothing useful to batch the test results.

Reply:
Sorry, it's bug.

Also fixed some other issues, e.g. in fact previously the newly added tests are not run, they still triggered TestRoundVectorFloatRandom.